inprocess/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -192,6 +203,7 @@ public synchronized Runnable start(ManagedClientTransport.Listener listener) {
         serverStreamTracerFactories = server.getStreamTracerFactories();
         // Must be semi-initialized; past this point, can begin receiving requests
         serverTransportListener = server.register(this);
+        serverAssumedMessageSize = server.getAssumedMessageSize();
```

Comment:
assigning the value is part of initialization, so should go before the comment above.  You want all initialization to be complete before registering it.